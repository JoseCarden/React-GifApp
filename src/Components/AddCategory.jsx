import { useState } from 'react';
import PropTypes from 'prop-types';

export const AddCategory = ({onNewCategory}) => {
  
  //Hook para controlar el valor que se está colocando en el
  //input
  const [inputValue, setInputValue] = useState('');

  //Recoge el evento de cambio en el input y toma al mismo (al 
  //txt/input) como el "target". De el se obtiene el valor para
  //que se haga un "seguimiento" a lo que se está tecleando
  const onInputChange = ({target}) =>
    setInputValue(target.value);

  const onSubmit = (event) =>{
    //evitar recarga de pagina tras el "enter"
    event.preventDefault();

    //evaluar q se halla más de un caracter
    if (inputValue.trim().length <= 1) return;

    //Devuelve el valor escrito sin espacios al final y inicio
    onNewCategory(inputValue.trim());

    //limpiar el input luego de presionar "enter"
    setInputValue('');
  }
  
  return (
    <form onSubmit={onSubmit} aria-label='form'>
        <input
            type="text"
            placeholder="Buscar gifs"
            value={inputValue}
            onChange={onInputChange}
        />
    </form>
  )
}

AddCategory.propTypes = {
  onNewCategory: PropTypes.func.isRequired
}
