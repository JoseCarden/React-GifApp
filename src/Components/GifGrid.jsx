import { GifItem } from './GifItem';
import { useFetchGifs } from '../hooks/useFetchGifs';
import PropTypes from 'prop-types';

export const GifGrid = ({category}) => {
    
    //Instancia de custom hook que trae las imagenes
    //y ayuda con la condicional para mostrar el mensaje
    // de "Cargando..." o no
    const {images, isLoading} = useFetchGifs(category);

    return (
    <>
        <h3>{category}</h3>

        {/*Evaluaci칩n de booleano que muestra un peque침o mensaje.
        Se puede trabajar de diferentes maneras, al menos lo que
        se puede mostrar como se침a de que est치 cargando */}
        {
            isLoading && (<h2>Cargando...</h2>)
        }       

        <div className="card-grid">
            {
            images.map( (image) => (
                <GifItem 
                    key={image.id}
                    {...image}
                />
                ) )
            }
        </div>
    </>
)
}

GifGrid.propTypes = {
    category: PropTypes.string.isRequired
}
