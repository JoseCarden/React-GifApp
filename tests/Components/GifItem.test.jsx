import { render } from "@testing-library/react";
import { GifItem } from '../../src/Components/GifItem'

describe('Pruebas en <GifItem/>', () => { 
    
    const title = 'Titulo';
    const url = 'https://www.youtube.com/watch?v=F8BMxX4E440&list=TLPQMTAwMjIwMjTcltn1vfZLhA&index=2';

    test('debe enviar titulo y url', () => { 
        render(<GifItem title={title} url= {url}/>)
    });

    test('debe ser igual al snapshot', () => { 

        const {container} = render(
            <GifItem 
                title={title}
                url= {url}
            />)
        expect(container).toMatchSnapshot();

     })

});
