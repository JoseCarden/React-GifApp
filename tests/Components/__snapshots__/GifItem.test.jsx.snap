// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pruebas en <GifItem/> debe ser igual al snapshot 1`] = `
<div>
  <div
    class="card"
  >
    <img
      alt="Titulo"
      src="https://www.youtube.com/watch?v=F8BMxX4E440&list=TLPQMTAwMjIwMjTcltn1vfZLhA&index=2"
    />
    <p>
      Titulo
    </p>
  </div>
</div>
`;
